### DISCUSSION TITLE
No `ToSql`/`FromSql` implementations for `chrono::DateTime<FixedOffset>` and `Timestamptz`?

### DISCUSSION THREAD

**Discussion by si14** [diesel-rs/diesel#4383] [Q&A] [answered]
Hey there!

I noticed that I can't store `chrono::DateTime<FixedOffset>` in Postgres with Diesel. I looked through all issues, discussions, and code, and somehow it seems that it was never a problem for anyone. Am I missing something obvious, or is it just an oversight? I can, theoretically, convert `FixedOffset` to `Utc`, but in my case I want to store the precise offset that was provided to me elsewhere, instead of erasing it by normalising to `Utc`. I hacked around with an rfc3339 `String`, but this is not ideal.

**Comment by weiznich** [comment, marked as answer]
> Am I missing something obvious, or is it just an oversight? I can, theoretically, convert FixedOffset to Utc, but in my case I want to store the precise offset that was provided to me elsewhere, instead of erasing it by normalising to Utc

Based on this comment I think you have a misunderstanding what `TIMESTAMPTZ` does:

> For timestamp with time zone, the internally stored value is always in UTC (Universal Coordinated Time, traditionally known as Greenwich Mean Time, GMT). An input value that has an explicit time zone specified is converted to UTC using the appropriate offset for that time zone. If no time zone is stated in the input string, then it is assumed to be in the time zone indicated by the system's [TimeZone](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-TIMEZONE) parameter, and is converted to UTC using the offset for the timezone zone.
> 
> When a timestamp with time zone value is output, it is always converted from UTC to the current timezone zone, and displayed as local time in that zone. To see the time in another time zone, either change timezone or use the AT TIME ZONE construct (see [Section 9.9.4](https://www.postgresql.org/docs/current/functions-datetime.html#FUNCTIONS-DATETIME-ZONECONVERT)).

[Source](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-INPUT-TIME-STAMPS)

To word this differently: Postgres itself is always normalizing these timestamps for UTC, which in turn is the reason why diesel exposes that in this particular way.

**Reply by si14** [reply]
Shame on me, I glanced over that section and didn't actually read it properly. Thank you, it makes total sense!
